HF technical accuracy audit: Staking & Snarking (#539)
o1-labs/docs2: docs/node-operators/staking-and-snarking.mdx

## Staking MINA

You can earn block rewards by participating in block production through staking. 

- Your new stake delegation comes into effect after a latency period of a 2-4 weeks.
- You can undelegate at any time with no penalty. Submit a delegation transaction with the new staking service or back to yourself if you want to be a block producer. Updates are made after a delay of 1-2 epochs.

### Requirements

To properly remain synced to the network and participate in consensus, your server must run a [clock synchronization protocol](https://en.wikipedia.org/wiki/Clock_synchronization).

Network Time Protocol [(NTP)](https://en.wikipedia.org/wiki/Network_Time_Protocol) is the most common method for synchronizing computer clocks over a network. 

To synchronize your server software clock with internet time servers, be sure to set up NTP. Many popular Linux distributions include NTP as a default service.

## Set or change the block producer key

> Use the `mina client delegate-stake` subcommand to set or change delegation of the block producer key. See the [Mina Client](/node-operators/mina-cli-reference#mina-client) CLI help.

Reviewer: I think it is better to point to the `Delegating MINA` that follows this section and which has more information about delegating

> Use the `mina client delegate-stake` subcommand to set or change delegation of the block producer key. See the [Mina Client](/node-operators/mina-cli-reference#mina-client) CLI help.

Author: ```suggestion
Use the `mina client delegate-stake` subcommand to set or change the delegation of the block producer key. See
 [Delegating MINA](#delegating-mina).
```

---
title: Staking and Snarking
hide_title: true
description: Mina Protocol provides block reward incentives to participate in block production through staking or delegation.
keywords:
  - proof of stake (PoS)
  - mina protocol
  - consensus mechanism
  - mina consensus
  - block production
  - incentives
  - staking
  - delegation
  - delegate
  - block rewards
---

# Staking and Snarking

> You can interact with the Mina network by participating in consensus and helping compress data by generating zk-SNARKs. By operating a node that helps secure the network, you can receive MINA for your efforts. See [Staking Rewards on Mina](https://minaprotocol.com/blog/staking-rewards-on-mina).

Reviewer 2: I would propose to change "compress" to something more like "produce a cryptographic proof for the data".

Compression is only a very imprecise analogy.

> You can interact with the Mina network by participating in consensus and helping compress data by generating zk-SNARKs. By operating a node that helps secure the network, you can receive MINA for your efforts. See [Staking Rewards on Mina](https://minaprotocol.com/blog/staking-rewards-on-mina).

Author: ```suggestion
You can interact with the Mina network by participating in consensus to produce a cryptographic proof for the data by generating zk-SNARKs. By operating a node that helps secure the network, you can receive MINA for your efforts. See [Staking Rewards on Mina](https://minaprotocol.com/blog/staking-rewards-on-mina).
```

Mina is a public and decentralized blockchain that is open for anyone in the world to participate in actively or passively. You can help increase the security of the network by becoming nodes or block producers, or you can help lower the cost of transactions by becoming SNARK producers, or you can be both.

## Staking MINA

You can earn block rewards by participating in block production through staking. Your new stake delegation comes into effect after a latency period of a 2-4 weeks.

### Requirements

To properly remain synced to the network and participate in consensus, your server must run a [clock synchronization protocol](https://en.wikipedia.org/wiki/Clock_synchronization).

Network Time Protocol [(NTP)](https://en.wikipedia.org/wiki/Network_Time_Protocol) is the most common method for synchronizing computer clocks over a network. 

To synchronize your server software clock with internet time servers, be sure to set up NTP. Many popular Linux distributions include NTP as a default service.

To stake with your MINA:

```sh

> mina client set-staking --public-key $MINA_PUBLIC_KEY

Author: protocol team, please verify that the set-staking subcommand is present in the command line help 
(or we can update the doc and instructions if the workflow is changed)

> Use the `mina client delegate-stake` subcommand to set or change the block producer key. See the [Mina Client](/node-operators/mina-cli-reference#mina-client) CLI help.

Author: great @georgeee  much clearer

Libp2p PeerID:              ...
```

where:

- The `Block producers running` field returns the number of accounts currently staking, with the associated key.

:::caution

When you are staking independently with funds in a account, you must remain connected to the network at all times to successfully produce blocks. If you need to go offline frequently, it might be better to delegate your stake to a node operator.

:::

If you want to send your coinbase to an account other than the one that is staking, use the `-coinbase-receiver` flag when you start your daemon. You can even point the coinbase at a cold wallet.

### Delegating MINA

> Delegating MINA has the benefit of not having to maintain a node that is always connected to the network.

Author: ```suggestion
Delegating MINA has the benefit of not having to maintain a node that is always connected to the network. See [Mina Foundation Delegation Program](/node-operators/foundation-delegation-program).
```
